fpmsyncd/routesync.cpp

```diff
@@ -661,6 +1537,13 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
 
     if (nlmsg_type == RTM_DELROUTE)
     {
+        char destipaddress[MAX_ADDR_SIZE + 1] = {0};
+        nl_addr2str(dip, destipaddress, MAX_ADDR_SIZE);
+
+        /* Duplicated delete as we do not know if it is a seg6 route, seg6local
+         * route or regular route */
+        m_srv6LocalSidTable.del(destipaddress);
```

Comment:
1) don't you need to check ip type to be v6 first?
2) How about handling warmRestartInProgress check just like is handled for normal route delete?
3) Same as earlier comments - srv6orch.cpp seems to be expecting more than just ip address as a key value

Reply:
I moved the delete code to `onSrv6LocalSidMsg()` and used the correct key value.